xds/internal/balancer/clusterresolver/balancerconfig/type_test.go

```diff
@@ -0,0 +1,94 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package balancerconfig
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestDiscoveryMechanismTypeMarshalJSON(t *testing.T) {
+	tests := []struct {
+		name string
+		typ  DiscoveryMechanismType
+		want string
+	}{
+		{
+			name: "eds",
+			typ:  DiscoveryMechanismTypeEDS,
+			want: `"EDS"`,
+		},
+		{
+			name: "dns",
+			typ:  DiscoveryMechanismTypeLogicalDNS,
+			want: `"LOGICAL_DNS"`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := json.Marshal(tt.typ)
+			if err != nil {
+				t.Errorf("DiscoveryMechanismTypeEDS.MarshalJSON() error = %v, wantErr %v", err, false)
+				return
+			}
+			if gotS := string(got); gotS != tt.want {
+				t.Errorf("DiscoveryMechanismTypeEDS.MarshalJSON() got %q, want %q", gotS, tt.want)
+			}
+		})
+	}
+}
+func TestDiscoveryMechanismTypeUnmarshalJSON(t *testing.T) {
+	tests := []struct {
+		name    string
+		js      string
+		want    DiscoveryMechanismType
+		wantErr bool
+	}{
+		{
+			name: "eds",
+			js:   `"EDS"`,
+			want: DiscoveryMechanismTypeEDS,
+		},
+		{
+			name: "dns",
+			js:   `"LOGICAL_DNS"`,
+			want: DiscoveryMechanismTypeLogicalDNS,
+		},
+		{
+			name:    "error",
+			js:      `"1234"`,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var got DiscoveryMechanismType
+			err := json.Unmarshal([]byte(tt.js), &got)
+			if (err != nil) != tt.wantErr {
```

Comment:
Combine the two?

Reply:
This is a little bit weird to combine (this prints diff).
I'm leaving them as separate checks, but using Fatalf